pycraigslist/exceptions.py

```diff
@@ -8,3 +8,13 @@
 
 class MaximumRequestsError(Exception):
     """Exceeds maximum get requests."""
+
+
+class InvalidFilterValue(ValueError):
+    """Search filter has no or an invalid value."""
+
+    def __init__(self, message, name, value):
```

Comment:
I don't think the constructor method is necessary for this case - could you remove it? Otherwise it looks great

Reply:
The constructor method is necessary so that the name of the filter and the associated incorrect value can be programmatically accessed.

The idea is that in code that uses pycraigslist, I can do something like the following:
```python
import pycraigslist

try:
    craigslist = pycraigslist.forsale.cta(site="sfbay", area="eby", query="Mazda Miata", auto_transmission=1)
except InvalidFilterValue as exc:
    print(f"Craigslist filter '{exc.name}' was provided an invalid value '{exc.value}'. Please correct this value and try to search again.")
else:
    for miata in craigslist.search():
        print(miata)
```
Basically, in my code that uses pycraigslist, a user could be able to pass N number of filters into my code, any of which could have a bad value. The point of this PR is that I want to be able to access the specific filter name and value that caused pycraigslist's initialization to fail so that I can customize the error message returned to the user. With the current implementation with ValueError, I'd have to access the message passed to ValueError and parse the string manually.

If there's another way to pass arguments into a custom exception for later access as an attribute that I don't know about, definitely let me know and I can learn about that (and refactor some other code I have to be more efficient! 😅)

Reply:
Okay, that makes sense -- thanks for the explanation